traits/trait_converters.py

```diff
@@ -32,14 +32,40 @@
 def trait_cast(obj):
     """ Convert to a CTrait if the object knows how, else return None.
     """
-    # special-case TraitType classes
-    if isinstance(obj, type) and hasattr(obj, 'instantiate_and_get_ctrait'):
-        return obj.instantiate_and_get_ctrait()
 
-    if hasattr(obj, 'as_ctrait'):
-        return obj.as_ctrait()
+    try:
```

Comment:
If you are making this change, you might look at making similar changes to `check_trait`, `trait_from` and `trait_for`.

Reply:
A straight forward change to these converters breaks tests in `test_rich_compare.py`. I created  a separate issue (#784) to track this.

Reply:
```
def check_trait(trait):
    """ Returns either the original value or a valid CTrait if the value can be
        converted to a CTrait.
    """
    try:
        value = as_ctrait(trait)
    except TypeError:
        value = trait

    return value
```

gives this error

```
FAIL: test_rich_compare_true (traits.tests.test_rich_compare.OldRichCompareTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/smohammed/devel/enthought/traits/traits/tests/test_rich_compare.py", line 238, in test_rich_compare_true
    self.assertIn(this_module, warn_msg.filename)
AssertionError: 'test_rich_compare' not found in '/home/smohammed/devel/enthought/traits/traits/has_traits.py'
```